WPThemeReview/Tests/Templates/ReservedTemplateNameTests/category-books.inc

```diff
@@ -0,0 +1,4 @@
+<?php
+/* Template Name: Books category template */
+
+// code goes here
```

Comment:
Might be an idea to not have a full PHP open tag in this file, but just a short echo open tag `<?=` somewhere in embedded PHP to verify that the sniff also works correctly when encountering templates without proper docblocks.

Reply:
Not sure I understand this. I need to have a `/* Template Name: Books category template */` in order for WP to recognise the custom templates, no?

```php
<?php
/* Template Name: Books category template */
?>

<div class="some-class-name">
  <?= 'This is an echo'; ?>
</div>
```

This now throws error about the template and the short echo used 😄

Reply:
From https://github.com/WPTRT/WPThemeReview/pull/213#pullrequestreview-236358438:

> > I'm not 100% sure if I should check for the comment at all, or just check for prefixed file names.
>
> I don't think you should as I don't think WP core does when including the file, so checking for the comments would lead to false negatives. (should be verified though, I haven't dug that deep)

In other words, it needs to be checked if WP Core checks the comment at the top of template files.
If it does, so should this sniff.
If it doesn't, this sniff shouldn't either.

Reply:
> This now throws error about the template and the short echo used

Correct. Sniffs should work independently of each other.

Reply:
So my example is correct?

Reply:
Unfortunately, no.

The sniff will only look at the file once, so with your code sample would get triggered on the first `<?php` and wouldn't get triggered anymore on the `<?=`, so the sniff being triggered on short open echo tags is still not being tested.

Reply:
So I should put the HTML part before the template comment? I've never seen such a file in a template 😕

Reply:
> So I should put the HTML part before the template comment? I've never seen such a file in a template

No, the test file I'm suggesting shouldn't have a (PHP) comment, just template code.
Something like:
```html
<!-- Template Name: Books category template -->
<div class="some-class-name">
  <?= 'This is an echo'; ?>
</div>
```